File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/ProjectSkillForm.ts
import { LookupEditor, IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ProjectSkillForm {
    SkillCategoryId: LookupEditor;
    SkillId: LookupEditor;
    ProjectId: IntegerEditor;
}

export class ProjectSkillForm extends PrefixedContext {
    static readonly formKey = 'Masters.ProjectSkill';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProjectSkillForm.init)  {
            ProjectSkillForm.init = true;

            var w0 = LookupEditor;
            var w1 = IntegerEditor;

            initFormType(ProjectSkillForm, [
                'SkillCategoryId', w0,
                'SkillId', w0,
                'ProjectId', w1
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Masters/ProjectSkill/ProjectSkillForm.cs
using Serenity.ComponentModel;

namespace GXpert.Masters.Forms;

[FormScript("Masters.ProjectSkill")]
[BasedOnRow(typeof(ProjectSkillRow), CheckNames = true)]
public class ProjectSkillForm
{
    public int SkillCategoryId { get; set; }
    public int SkillId { get; set; }
    public int ProjectId { get; set; }
}